dds/DCPS/DataReaderImpl.cpp

```diff
@@ -1429,9 +1425,9 @@ DataReaderImpl::data_received(const ReceivedDataSample& sample)
     bool is_new_instance = false;
     bool filtered = false;
     if (sample.header_.key_fields_only_) {
-      dds_demarshal(sample, publication_handle, instance, is_new_instance, filtered, KEY_ONLY_MARSHALING, false);
+      dds_demarshal(sample, publication_handle, instance, is_new_instance, filtered, KEY_ONLY_MARSHALING);
     } else {
-      real_data = dds_demarshal(sample, publication_handle, instance, is_new_instance, filtered, FULL_MARSHALING, observer && vd);
+      dds_demarshal(sample, publication_handle, instance, is_new_instance, filtered, FULL_MARSHALING);
```

Comment:
Since these statements are essentially the same, collapse them down to one (either a named variable or a `? :` would work well for the one different parameter).